bot/cogs/memes_cog/memes_cog.py

```diff
@@ -95,5 +103,51 @@ async def spongebob(self, ctx, *, args):
 
         await ctx.send(embed=embed)
 
+    @commands.command(aliases=['rave', '🦀'])
+    async def crab(self, ctx, *, args='Bottom text\n is dead'):
+        # crab.gif dimensions - 352 by 200
+        # Immediately grab the timestamp incase of multiple calls in a row
+        timestamp = datetime.datetime.utcnow()
+
+        # Open crab.gif and add our font
+        im = Image.open('bot/cogs/memes_cog/assets/crab.gif')
+        fnt_path = 'bot/cogs/memes_cog/assets/LemonMilk.otf'
+        fnt = ImageFont.truetype(fnt_path, 11)
+
+        # Add new lines for when the text would go out of bounds
+        lines_in_text = 1
+        while len(args) > (crab_line_length * lines_in_text):
+            newline_loc = crab_line_length * lines_in_text
+            # I didn't want to add a newline in the middle of a word
+            while not args[newline_loc].isspace():
+                newline_loc -= 1
+            args = f'{args[:newline_loc]} \n{args[newline_loc:]}'
+            lines_in_text += 1
+        
+        # Draw the text on to each frame of the gif
+        # Gonna be honest I don't quite understand how it works but I got it from the Pillow docs/issues
+        frames = []
+        for frame in ImageSequence.Iterator(im):
+            d = ImageDraw.Draw(frame)
+            w, h = d.textsize(args, fnt)
+            # draws the text on to the frame. Tries to center horizontally and tries to go as close to the bottom as possible
+            d.text((im.size[0]/2 - w/2, im.size[1] - h - (5 * lines_in_text)), args, font=fnt, align='center', stroke_width=1, stroke_fill=Colors.ClemsonOrange, spacing=6)
+            del d
+
+            b = io.BytesIO()
+            frame.save(b, format='GIF')
+            frame = Image.open(b)
+            frames.append(frame)
+
+        # Save, send, and delete created gif
+        frames[0].save(f'bot/cogs/memes_cog/assets/out_{timestamp}.gif', save_all=True, append_images=frames[1:])
+        attachment = discord.File(filename=f'out_{timestamp}.gif', fp=f'bot/cogs/memes_cog/assets/out_{timestamp}.gif')
+        msg = await ctx.send('Generating your gif') 
```

Comment:
Lets move this to the top of the command, in my testing the pillow stuff wasnt a super insignificant amount of time